### PR TITLE
CleanUp

### PR DESCRIPTION
**Pull request by beauremus**
Leverage the features added in #7 like type hinting and test parameterization. Also, clean up flake8 issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,11 +126,27 @@ To test local modifications, use pip's editable mode.
 To run unit tests under `test` folder:
 
 ```bash
-coverage run -m unittest discover
+make test
```
Comment (julianbadillo) [inline README.md:131]: I have to use `pymake` in Windows. <https://pypi.org/project/py-make/>

Comment (beauremus) [reply README.md:131]: I didn't think about that. Should we find another tool? I think the standard option is to run commands by hand.

File: tests/drf_generator_test.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+import drf_generator.generator as drf_generator
+
+parent_path = Path(__file__).parent
+first_validation_file = parent_path.joinpath('test_validate1.txt').resolve()
+second_validation_file = parent_path.joinpath(
+    'test_validate2.txt').resolve()
+
+
+def equal(first, second):
```
Comment (julianbadillo) [inline tests/drf_generator_test.py:18]: Wrapping an operator in a function looks like an overkill to me. A lambda might suffice, but then even the operator itself would be enough.

File: tests/drf_generator_test.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    Test the char_range function.
+
+    :param first: First character.
+    :param second: Second character.
+    :param expected: Expected result.
+    :param comparator: Comparator function.
+    """
+    result = list(drf_generator.char_range(first, second))
+    assert comparator(result, expected)
+
```
Comment (julianbadillo) [inline tests/drf_generator_test.py:46]: Why isn't just
```python
assert result == expected
```
I see you'd want to leave room for testing more than one comparator, but it just makes harder to figure out what the test is actually testing, and in failure case, it's a bit convoluted to see the error.
If we want to make another type of list comparison (like inclusion, or permutation), it's ok to write another test case. IMO it's an unneeded abstraction.

**Review by julianbadillo — COMMENTED** [review]
Though it's a clever trick, I like the unit tests better without the `comparator` parameter, I had a hard time articulating why beyond _"it looks ugly"_, but these folks sum it up ok: <https://www.nerdwallet.com/blog/engineering/5-pytest-best-practices/>
Bottom line, a test case should test one behavior; **different behaviors should go on different test cases**.
They say something similar here: <https://realpython.com/pytest-python-testing/#parametrization-combining-tests>

> _Even though this shortened your code, it’s important to note that in this case, it didn’t do much to clarify your test code. Use parametrization to separate the test data from the test behavior so that it’s clear what the test is testing!_

File: README.md (hunk 1 above)
Comment (beauremus) [reply README.md:131]: Should we instead use a containerized development environment to simplify accounting for various operating system development tools?

File: tests/drf_generator_test.py (hunk 2 above)
Comment (beauremus) [reply tests/drf_generator_test.py:18]: We can just pass `==`!? That'd be awesome!

File: README.md (hunk 1 above)
Comment (julianbadillo) [reply README.md:131]: Nah, not worth it, I was reading around and this is pretty standard practice for python development, plus the workflow works fine on Windows. I'm happy if you just make a side note to use `pymake` for Win.

File: tests/drf_generator_test.py (hunk 2 above)
Comment (beauremus) [reply tests/drf_generator_test.py:18]: I don't think I interpretted what you are saying correctly. What do you mean by "the operator itsefl would be enough"?

**Pull request merged**